src/OpenSage.Mods.Generals/Gui/SkirmishGameOptionsMenuCallbacks.cs

```diff
@@ -35,5 +44,61 @@ public static void SkirmishGameOptionsMenuSystem(Control control, WndWindowMessa
                     break;
             }
         }
+
+        public static void SkirmishGameOptionsMenuInit(Window window, Game game)
+        {
+
+            _window = window;
+            _game = game;
+
+            var playerNameTextBox =  (TextBox)_window.Controls.FindControl("SkirmishGameOptionsMenu.wnd:TextEntryPlayerName");
+            playerNameTextBox.IsReadOnly = false;
+            
+
+            FillComboBoxOptions(ComboBoxTeamPrefix, new[]
+            {
+                "Team:0", "Team:1", "Team:2", "Team:3", "Team:4"
+            });
+
+            var playableSides = game.ContentManager.IniDataContext.PlayerTemplates.FindAll(i => i.PlayableSide);
+            if(playableSides.Count > 0)
+            {
+                var sideList = playableSides.Select(i => i.DisplayName).ToList();
+                sideList.Insert(0, "GUI:RandomSide");
+
+                FillComboBoxOptions(ComboBoxPlayerTemplatePrefix, sideList.ToArray());
+            }
+            
+            //Maybe we can selcet the data by linq with prefix Color:
+            FillComboBoxOptions(ComboBoxColorPrefix, new[]
+            {
+                "GUI:???", "Color:Gold", "Color:Red", "Color:Green", "Color:Orange", "Color:SkyBlue", "Color:Purple", "Color:Pink"
+            });
+
+            FillComboBoxOptions(ComboBoxPlayerPrefix, new[]
+            {
+                "GUI:Open", "GUI:Closed", "GUI:EasyAI", "GUI:MediumAI", "GUI:HardAI"
+            });
+
+        }
+
+        /// <summary>
+        /// Fill all related ComboBoxs with the same data
+        /// </summary>
+        /// <param name="key"></param>
+        /// <param name="options"></param>
+        /// <param name="selectedIndex"></param>
+        private static void FillComboBoxOptions(string key, string[] options, int selectedIndex = 0)
+        {
+            var comboBoxs = _window.Controls.FindControlsStratsWith<ComboBox>(key);
+            foreach (ComboBox comboBox in comboBoxs)
+            {
+                if(comboBox.Name.Length -1 != key.Length) continue;
```

Comment:
This is currently needed because we have two ComboBox names that start identically. Here is checked for exact length -1. So that only one number can come after the key.

"SkirmishGameOptionsMenu.wnd:ComboBoxPlayerTemplate";
"SkirmishGameOptionsMenu.wnd:ComboBoxPlayer";